android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt

```diff
@@ -190,6 +193,75 @@ class OkHttpTest {
     }
   }
 
+  @Test
+  fun testEventListener() {
+    val eventListener = RecordingEventListener()
+
+    client = client.newBuilder().eventListener(eventListener).build()
+
+    enableTls()
+
+    server.enqueue(MockResponse().setBody("abc1"))
+    server.enqueue(MockResponse().setBody("abc2"))
+
+    val request = Request.Builder().url(server.url("/")).build()
+
+    client.newCall(request).execute().use { response ->
+      assertEquals(200, response.code)
+    }
+
+    assertEquals(listOf("CallStart", "ProxySelectStart", "ProxySelectEnd", "DnsStart", "DnsEnd",
+        "ConnectStart", "SecureConnectStart", "SecureConnectEnd", "ConnectEnd",
+        "ConnectionAcquired", "RequestHeadersStart", "RequestHeadersEnd", "ResponseHeadersStart",
+        "ResponseHeadersEnd", "ResponseBodyStart", "ResponseBodyEnd", "ConnectionReleased",
+        "CallEnd"), eventListener.recordedEventTypes())
+
+    eventListener.clearAllEvents()
+
+    client.newCall(request).execute().use { response ->
+      assertEquals(200, response.code)
+    }
+
+    assertEquals(listOf("CallStart", "ProxySelectStart", "ProxySelectEnd",
+        "ConnectionAcquired", "RequestHeadersStart", "RequestHeadersEnd", "ResponseHeadersStart",
+        "ResponseHeadersEnd", "ResponseBodyStart", "ResponseBodyEnd", "ConnectionReleased",
+        "CallEnd"), eventListener.recordedEventTypes())
+  }
+
+  @Test
+  fun testSessionReuse() {
+    val sessionIds = mutableListOf<String>()
+
+    client = client.newBuilder().eventListener(object : EventListener() {
+      override fun connectionAcquired(call: Call, connection: Connection) {
+        val sslSocket = connection.socket() as SSLSocket
+
+        sessionIds.add(sslSocket.session.id.toByteString().hex())
+      }
+    }).build()
+
+    enableTls()
+
+    server.enqueue(MockResponse().setBody("abc1"))
+    server.enqueue(MockResponse().setBody("abc2"))
+
+    val request = Request.Builder().url(server.url("/")).build()
+
+    client.newCall(request).execute().use { response ->
+      assertEquals(200, response.code)
+    }
+
+    client.connectionPool.evictAll()
+    assertEquals(0, client.connectionPool.connectionCount())
+
+    client.newCall(request).execute().use { response ->
+      assertEquals(200, response.code)
+    }
+
+    assertEquals(2, sessionIds.size)
+    assertEquals(sessionIds[0], sessionIds[1])
```

Comment:
Sweet test.

Reply:
Hi @yschimke, I am trying this session reuse test with Android 7.1.1 but it is not working for me. 
What version of Android is compatible with session reuse?

Reply:
I think that's a question for https://github.com/google/conscrypt